File: rockets/src/main/kotlin/com/spacex/rockets/datasources/local/RocketsDaoModule.java
package com.spacex.rockets.datasources.local;

import android.content.Context;
import androidx.annotation.NonNull;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class RocketsDaoModule {

  @Provides
  static RocketsDao provideRocketsDao(@NonNull final Context context) {
    return RocketsDatabase.getInstance(context).getRocketsDao();
  }
}

File: rockets/src/main/kotlin/com/spacex/rockets/datasources/local/RocketsDaoModule.kt
package com.spacex.rockets.datasources.local

import android.content.Context
import androidx.room.Room
import dagger.Module
import dagger.Provides

@Module
internal object RocketsDaoModule {

    @JvmStatic
    @Provides
    fun provideRocketsDao(context: Context): RocketsDao =
            Room.databaseBuilder(
                    context.applicationContext, RocketsDatabase::class.java, "Rockets Database")
                    .fallbackToDestructiveMigration()
                    .build().rocketsDao
}
